packages/blade/src/components/Radio/useRadio.ts

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import React from 'react';
+import type { GestureResponderEvent } from 'react-native';
+import type { RadioProps } from './Radio';
+import { useControllableState } from '~src/hooks/useControllable';
+import { useId } from '~src/hooks/useId';
+import { getPlatformType, makeAccessible } from '~utils';
+
+type UseRadioProps = Pick<
+  RadioProps,
+  'isChecked' | 'defaultChecked' | 'isDisabled' | 'isRequired' | 'onChange' | 'name' | 'value'
+> & { hasError?: boolean };
+
+const useRadio = ({
+  isChecked,
+  defaultChecked,
+  isDisabled,
+  isRequired,
+  hasError,
+  onChange,
+  name,
+  value,
+}: UseRadioProps) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const isReactNative = getPlatformType() === 'react-native';
```

Comment:
Can we move this logic to appropriate .web / .native?

Reply:
Again, not feasible. 
Imagine maintaining useRadio twice with identical logic. 

.web / .native files makes more sense when there's little to know overlap or lets say when we use the Styled* components.